src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -788,6 +811,10 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
         }
     }
 
+    pub fn check_profile(&self, _pkg: &PackageId) -> &'a Profile {
```

Comment:
It's ok to avoid a new method here, `lib_profile` was added a point I thought I would use it but never ended up doing so.